### DISCUSSION TITLE
ANSYS install used by DPF

### DISCUSSION THREAD

**Discussion by agrishin** [ansys/pydpf-core#1280] [Q&A] [answered]
From Costas Vogiatzis at Honeywell:
On a computer with multiple ANSYS installations, I am running a Python script that uses DPF. As I understand it, DPF uses some DLLs from the ANSYS installation. From within that script, how can I find which of the ANSYS installation DPF is using?

**Comment by mikerife** [comment]
Hi @agrishin 
You should ask this on the PyDPF GitHub site instead of here.  https://github.com/ansys/pydpf-core
Since there is not a 1-to-1 correlation between releases of Ansys products and what they support with respect to PyDPF then when started the PYDPF server will try the latest version of Ansys install and if that is not available, work its way to the next etc.  There used to be an environment variable that could be used to define which one Ansys install to use; I think it may have been deprecated due to changes to the way DPF works.  

Mike

**Comment by germa89** [comment]
I transferred the discussion.


Pinging @PProfizi @cbellot000

**Comment by PProfizi** [comment, marked as answer]
Hi @agrishin,

To know which ANSYS installation your PyDPF script is going to use, you can try out the following script which starts a server with the default logic and then prints out its version and path where DLLs will be used:
```
import ansys.dpf.core as dpf

server = dpf.start_local_server()
print(server.version)
print(server.ansys_path)
```

That makes me think I should wrap this in a one-liner for easier use, as well as in a "diagnostic" tool which would give a summary of everything going on.

The environment variable mentioned by @mikerife is `ANSYS_DPF_PATH`. It can be used to set a default ANSYS install to use anytime DPF tries to start a server. 

What you can otherwise do is define it at the start of your script by calling [ansys.dpf.core.start_local_server()](https://dpf.docs.pyansys.com/version/stable/api/ansys.dpf.core.server.html#ansys.dpf.core.server.start_local_server) with argument `ansys_path`.

**Reply by PProfizi** [reply]
See https://dpf.docs.pyansys.com/version/stable/getting_started/dpf_server.html#manage-multiple-dpf-server-installations
